### PR TITLE
YARN-11524. Improve the Policy Description in Federation.md.

### PR DESCRIPTION
**Pull request by slfan1989**
### Description of PR
JIRA: YARN-11524. Improve the Policy Description in Federation.md.

The `Federation.md` file lacks an introduction to Policy. In this pr we will enhance the content regarding `Router Policy`, `AMRM Policy`, and `PolicyManager` to help users gain a better understanding of these types of policies.

- Router Policy
image

- AMRM Policy
image

- Policy Manager
image

### How was this patch tested?

Documentation modifications do not require unit tests.

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  46m 39s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 32s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  83m 13s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 17s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/1/artifact/out/blanks-eol.txt) |  The patch has 11 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  mvnsite  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m 28s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 122m 15s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 12088632f39d 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 002c574f1351dbd7d086447e01a87844abaeb3db |
| Max. process+thread count | 650 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 39s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  46m  9s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 35s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  80m 39s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m 18s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 119m  2s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 2caa612ad7ba 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / f3e4756afd2ed3909c89d469dd0bc557b0f7f254 |
| Max. process+thread count | 555 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  50m 46s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  84m  1s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m  2s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   | 122m 24s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux d1f4246e2608 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a501d5bc8469e2628fb4940f5f341d59d54a9967 |
| Max. process+thread count | 561 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/4/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 39s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  54m 25s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  88m 16s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 18s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m  9s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 38s |  |  The patch does not generate ASF License warnings.  |
|  |   | 126m 20s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 88174e672c60 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a501d5bc8469e2628fb4940f5f341d59d54a9967 |
| Max. process+thread count | 671 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md [hunk 1]
```diff
@@ -235,7 +235,112 @@ SQL-Server scripts are located in **sbin/FederationStateStore/SQLServer/**.
 |`yarn.federation.subcluster-resolver.class` | `org.apache.hadoop.yarn.server.federation.resolver.DefaultSubClusterResolverImpl` | The class used to resolve which subcluster a node belongs to, and which subcluster(s) a rack belongs to. |
 |`yarn.federation.machine-list` | `<path of machine-list file>` | Path of machine-list file used by `SubClusterResolver`. Each line of the file is a node with sub-cluster and rack information. Below is the example: <br/> <br/> node1, subcluster1, rack1 <br/> node2, subcluster2, rack1 <br/> node3, subcluster3, rack2 <br/> node4, subcluster3, rack2 |
 
-###ON RMs:
+**How to Configure policy-manager ?**
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: > How to configure the policy-manager?

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md [hunk 2]
```diff
@@ -235,7 +235,112 @@ SQL-Server scripts are located in **sbin/FederationStateStore/SQLServer/**.
...
+    - We configure to use this policy:
+    ```xml
+      <property>
+         <name>yarn.federation.policy-manager</name>
+         <value>org.apache.hadoop.yarn.server.federation.policies.manager.UniformBroadcastPolicyManager</value>
+      </property>
+    ```
+  - WeightedLocalityPolicyManager
+    - Policy that allows operator to configure "weights" for routing. This picks a LocalityRouterPolicy for the router and a LocalityMulticastAMRMProxyPolicy for the amrmproxy as they are designed to work together.
+    - We configure to use this policy:
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: This setting is a little repetitive as we say the same over and over.
Probably we can just give it at the beginning and refer to the name.

Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: Thank you for your suggestion! I will improve the documentation.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  52m  5s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  86m  3s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 18s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/5/artifact/out/blanks-eol.txt) |  The patch has 2 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m  0s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 124m 25s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux c5328f17198d 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 886cf54be5218a154216ade8a4d24e536fb2f6db |
| Max. process+thread count | 737 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/5/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 53s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  44m 39s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 35s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  78m 15s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  32m 54s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 116m 22s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 4ad6f8f28dda 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / f08139f275773ca33fe5b2214c88bac8a0c2c3f1 |
| Max. process+thread count | 555 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/6/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by slfan1989** [comment]
@goiri Can you help review this pr? Thank you very much!

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md [hunk 3]
```diff
@@ -349,6 +430,10 @@ Optional:
 | `yarn.federation.statestore.max-connections` | `1` | The maximum number of parallel connections from each AMRMProxy to the state-store. This value is typically lower than the router one, since we have many AMRMProxy that could burn-through many DB connections quickly. |
 | `yarn.federation.cache-ttl.secs` | `300` | The time to leave for the AMRMProxy cache. Typically larger than at the router, as the number of AMRMProxy is large, and we want to limit the load to the centralized state-store. |
 
+### ON GPG:
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: Nothing here?

Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: Thank you very much for your help in reviewing the code! `GPG` will submit pr separately to improve the description of md.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 53s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| -1 :x: |  mvninstall  |  44m 55s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/7/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  78m 27s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 18s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  32m 46s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   | 116m 43s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux 6fc4847b4160 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 63cb91bcf615c724c6d52f2cb95b8bb5d7043783 |
| Max. process+thread count | 555 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/7/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 56s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| -1 :x: |  mvninstall  |  45m 44s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/8/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  mvnsite  |   0m 36s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  79m  2s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m 23s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 118m  0s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux a1a0925388d6 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 9e3fda6aad82f99b38029e417e3fe74e81216f0c |
| Max. process+thread count | 706 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/8/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  2s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  47m 20s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 32s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  83m 55s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  32m 58s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 122m 31s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5797 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux a3da1174a81b 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / e95c90b4bdf1cf0f4267e3abe9ccc0afcb7002dd |
| Max. process+thread count | 555 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5797/9/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by slfan1989** [comment]
@goiri Can you help to review this PR again? Thank you very much!

**Review by goiri — APPROVED** [review]

**Comment by slfan1989** [comment]
@goiri Thank you very much for your help in reviewing the code!

**Pull request merged**